### PR TITLE
AUDIO: MT32: Update MT32 emulation code to mt32emu 2.7.0

### PR DESCRIPTION
**Pull request by lotharsm**
This patch (finally) updates the MT32 emulation code to mt32emu 2.7.0.

No warnings in GCC. I re-applied our patch for silencing MSVC warnings, but unfortunately, I can't test them right now.

Submitting this as PR since it touches all ports.

### CODE REVIEW

File: audio/softsynth/mt32/Analog.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* Copyright (C) 2003, 2004, 2005, 2006, 2008, 2009 Dean Beeler, Jerome Fisher
- * Copyright (C) 2011-2021 Dean Beeler, Jerome Fisher, Sergey V. Mikayev
+ * Copyright (C) 2011-2022 Dean Beeler, Jerome Fisher, Sergey V. Mikayev
```
Comment (bluegr) [inline audio/softsynth/mt32/Analog.cpp:2]: Since these changes produce a lot of noise, can you put them in a separate commit?

File: audio/softsynth/mt32/Analog.cpp [hunk 2]
```diff
@@ -223,23 +223,23 @@ class AnalogImpl : public Analog {
...
-	bool process(FloatSample *outStream, const FloatSample *nonReverbLeft, const FloatSample *nonReverbRight, const FloatSample *reverbDryLeft, const FloatSample *reverbDryRight, const FloatSample *reverbWetLeft, const FloatSample *reverbWetRight, Bit32u outLength) override;
+	bool process(IntSample *outStream, const IntSample *nonReverbLeft, const IntSample *nonReverbRight, const IntSample *reverbDryLeft, const IntSample *reverbDryRight, const IntSample *reverbWetLeft, const IntSample *reverbWetRight, Bit32u outLength);
+	bool process(FloatSample *outStream, const FloatSample *nonReverbLeft, const FloatSample *nonReverbRight, const FloatSample *reverbDryLeft, const FloatSample *reverbDryRight, const FloatSample *reverbWetLeft, const FloatSample *reverbWetRight, Bit32u outLength);
 
 	template <class Sample>
 	void produceOutput(Sample *outStream, const Sample *nonReverbLeft, const Sample *nonReverbRight, const Sample *reverbDryLeft, const Sample *reverbDryRight, const Sample *reverbWetLeft, const Sample *reverbWetRight, Bit32u outLength) {
-		if (outStream == nullptr) {
+		if (outStream == NULL) {
```
Comment (bluegr) [inline audio/softsynth/mt32/Analog.cpp:242]: Please don't replace nullptr (which is C++ style) with NULL (which is C style).

Check here why NULL was replaced with nullptr:
https://stackoverflow.com/questions/20509734/null-vs-nullptr-why-was-it-replaced

Comment (lotharsm) [reply audio/softsynth/mt32/Analog.cpp:242]: I didn't do the modifications on my own - I took the code directly from mt32emu (as always)...

Comment (lotharsm) [reply audio/softsynth/mt32/Analog.cpp:242]: I wonder what is the better approach here: modify the sources we import (and deviate from the original) or keeping the imported code as-is, "trusting" the original developers? I don't think they made this change without a reason though...

Comment (eriktorbjorn) [reply audio/softsynth/mt32/Analog.cpp:242]: Looks like upstreams always used NULL, and we changed it in https://github.com/scummvm/scummvm/commit/af529f568baa1f9e47a957e84fd17561c15613e0

**Comment by lotharsm** [comment]
> Looks like upstreams always used NULL, and we changed it in af529f5 
> <https://github.com/scummvm/scummvm/commit/af529f568baa1f9e47a957e84fd17561c15613e0>
> 

Yikes, looks like this slipped through. Should we re-apply this change 
or should we stick with upstream?

**Comment by lotharsm** [comment]
Yikes, I accidentally pushed the code to master via d94505d46e9226210ede4ec9e63f091270273b09.

Should we revert this or is it acceptable as-is, including the "untouched" sources?

**Comment by lotharsm** [comment]
Closing, fixing this in-tree if necessary.
